### ISSUE TITLE
Compiling with python3.10 fails

### ISSUE THREAD

**Issue by dschwoerer** [pysam-developers/pysam#1015]
pysam is using non-documented CPython API that is changed in python 3.10, see https://bugs.python.org/issue43760

The issue was first reported here https://bugzilla.redhat.com/show_bug.cgi?id=1958992

```
gcc -pthread -Wno-unused-result -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-p
rotector-strong -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIF
Y_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -fexceptions -g -grecord-gcc-switches -pi
pe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv
 -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-st
rong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fPIC -I/builddir/build/BUILD/pysam-0.16.0.1/htslib -I/builddir/build/BUILD/pysam-0.
16.0.1/samtools -I/builddir/build/BUILD/pysam-0.16.0.1/samtools/lz4 -I/builddir/build/BUILD/pysam-0.16.0.1/bcftools -I/builddir/build/BUILD/pysam-0.16.0.1/pysam -I/builddir/build/BUILD/pysam-0.16.0.1 -I/usr/include/python3.10
 -c pysam/libchtslib.c -o build/temp.linux-x86_64-3.10/pysam/libchtslib.o -Wno-unused -Wno-strict-prototypes -Wno-sign-compare -Wno-error=declaration-after-statement
pysam/libchtslib.c: In function ‘__Pyx_call_return_trace_func’:
pysam/libchtslib.c:1273:15: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘use_tracing’; did you mean ‘tracing’?
 1273 |       tstate->use_tracing = 0;
      |               ^~~~~~~~~~~
      |               tracing
pysam/libchtslib.c:1279:15: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘use_tracing’; did you mean ‘tracing’?
 1279 |       tstate->use_tracing = 1;
      |               ^~~~~~~~~~~
      |               tracing
...
```

**Comment by kevinjacobs-progenity** [comment]
I suspect this is an issue with Cython using the deprecated API.  Thanks for letting us know.

**Comment by jmarshall** [comment]
This has now been fixed in Cython, and the fix should appear in the upcoming Cython 0.29.24.
